iceoryx_examples/automotive_soa/include/topic.hpp

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_EXAMPLES_AUTOMOTIVE_SOA_TOPIC_HPP
+#define IOX_EXAMPLES_AUTOMOTIVE_SOA_TOPIC_HPP
+
+#include "iceoryx_hoofs/cxx/serialization.hpp"
+
+#include <chrono>
+#include <cstdint>
+
+constexpr uint32_t ONE_KILOBYTE = 1000U;
+
+struct AddRequest
+{
+    uint64_t addend1{0};
+    uint64_t addend2{0};
+};
+
+struct AddResponse
+{
+    uint64_t sum{0};
+};
+
+struct Topic
+{
+    uint32_t counter{0};
+};
+
+struct TimestampTopic1Byte
+{
+    // Printed to console
+    uint32_t counter{0};
+    std::chrono::time_point<std::chrono::steady_clock> sendTimestamp;
+
+    // Not printed to console
+    static constexpr uint32_t payloadSizeInBytes{1};
+    char data[payloadSizeInBytes];
+    uint32_t subPackets{0};
+};
+
+struct TimestampTopic4Kb
+{
+    // Printed to console
+    uint32_t counter{0};
+    std::chrono::time_point<std::chrono::steady_clock> sendTimestamp;
+
+    // Not printed to console
+    static constexpr uint32_t payloadSizeInBytes{4 * ONE_KILOBYTE};
```

Comment:
Why not make this a template. Then we could safe roughly 100 lines of code.
Just write:
```cxx
template<uint64_t Size>
struct TimestampTopic {
    // Printed to console
    uint32_t counter{0};
    std::chrono::time_point<std::chrono::steady_clock> sendTimestamp;

    // Not printed to console
    static constexpr uint32_t payloadSizeInBytes{Size};
    char data[Size];
    uint32_t subPackets{0};
}

using TimestampTopic4Kb = TimestampTopic<4 * ONE_KILOBYTE>;
```

This would really improve the readability. Otherwise one may wonder if those types may differ in something more then just the size but with this we clearly state the intend.

Reply:
I know what you mean. However, if you don't mind I would to keep the separate structs to make it explicit which data sizes are used for comparison of iceoryx and UNIX domain sockets. `is_supported_topic` should give an idea about how to implement a compatible type.